### ISSUE TITLE
[ROS 2]  Lidar Sensor does not allow for raycaster-side publication. 

### ISSUE THREAD

**Issue by alek-kam-robotec-ai** [o3de/o3de-extras#719]
**Describe the bug**
The pointcloud is always published in the ROS2 Lidar Sensor regardless of the configuration.
This is due to the fact that the raycasters are queried about their capability to publish before their configuration is finalized, namely
the `ConfigurePointCloudPublisher` is used after a call to `CanHandlePublishing` (methods of the LidarRequests interface) which requires the lidar to be already configured.

**Expected behavior**
The pointcloud publisher should be determined based on whether or not the Lidar implementation supports publishing under current lidar configuration.

**Found in Branch**
All branches. Was introduced by [commit](https://github.com/o3de/o3de-extras/commit/b5ce30ecd0900daa9c0d7914d47da844fe4d4912).

**Comment by alek-kam-robotec-ai** [comment]
@adamdbrw @michalpelka 
I've spoken with one of the RGL team members and we suggest that this is patched after the instance segmentation (#713) and intensity (under development) features are merged.

**Comment by jhanca-robotecai** [comment]
Isn't this one related to: https://github.com/o3de/o3de-extras/issues/462
and being fixed in https://github.com/o3de/o3de-extras/pull/724
?

**Comment by alek-kam-robotec-ai** [comment]
No, it is not. As I've described, it is related to the raycaster configuration and query order.

**Comment by alek-kam-robotec-ai** [comment]
Additionally, the same pull request hard-sets requested result types to both points and distances irrespective of the Lidar Component, which is not the intended functionality.
